app/src/main/java/com/infomaniak/swisstransfer/workers/UploadWorker.kt

```diff
@@ -0,0 +1,143 @@
+            chunkMinSize = EXPECTED_CHUNK_SIZE,
+            chunkMaxSize = EXPECTED_CHUNK_SIZE,
+            totalChunks = TOTAL_CHUNKS,
+        )
+    }
+    private val uploadFileTask by lazy {
+        UploadFileTask(uploadManager, fileChunkSizeManager)
+    }
+
+    private var totalUploadedBytes = 0L
+
+    override suspend fun launchWork(): Result {
+        SentryLog.i(TAG, "Work launched")
+
+        if (uploadManager.getUploadsCount() == 0L) {
+            SentryLog.w(TAG, "No upload pending")
+            return Result.failure()
+        }
+
+        val uploadSession = uploadManager.initUploadSession(recaptcha = "Recaptcha")!!
+        val totalSize = uploadSession.files.sumOf { it.size }
+
+        SentryLog.d(TAG, "Work started with ${uploadSession.files.count()} files of $totalSize bytes")
+
+        uploadSession.files.forEach { fileSession ->
+            uploadFileTask.start(fileSession, uploadSession) { bytesSent ->
+                totalUploadedBytes += bytesSent
+                setProgress(workDataOf(UPLOADED_BYTES_TAG to totalUploadedBytes, TOTAL_SIZE_TAG to totalSize))
+            }
+        }
+
+        uploadManager.finishUploadSession(uploadSession.uuid)
+        importLocalStorage.removeImportFolder()
+
+        return Result.success()
+    }
+
+    override fun onFinish() {
+        SentryLog.i(TAG, "Work finished")
+    }
+
+    @Singleton
+    class Scheduler @Inject constructor(private val workManager: WorkManager) {
+
+        fun scheduleWork() {
+            SentryLog.i(TAG, "Work scheduled")
+            val workRequest = OneTimeWorkRequestBuilder<UploadWorker>()
+                .setConstraints(Constraints.Builder().setRequiredNetworkType(NetworkType.CONNECTED).build())
+                .build()
+            workManager.enqueueUniqueWork(TAG, ExistingWorkPolicy.REPLACE, workRequest)
+        }
+
+        @OptIn(ExperimentalCoroutinesApi::class)
+        fun trackUploadProgressFlow(): Flow<UploadTransferProgress> {
+            val workQuery = WorkQuery.Builder.fromUniqueWorkNames(listOf(TAG))
+                .addStates(listOf(WorkInfo.State.RUNNING))
+                .build()
+            return workManager.getWorkInfosFlow(workQuery).mapLatest {
+                val workProgress = it.firstOrNull()?.progress ?: return@mapLatest null
+                UploadTransferProgress(
+                    uploadedSize = workProgress.getLong(UPLOADED_BYTES_TAG, 0L),
+                    totalSize = workProgress.getLong(TOTAL_SIZE_TAG, 0L),
+                )
+            }.filterNotNull()
+        }
+
+        @OptIn(ExperimentalCoroutinesApi::class)
+        fun isPendingOrRunningFlow(): Flow<Boolean> {
+            val workQuery = WorkQuery.Builder.fromUniqueWorkNames(listOf(TAG))
+                .addStates(listOf(WorkInfo.State.RUNNING, WorkInfo.State.ENQUEUED, WorkInfo.State.BLOCKED))
+                .build()
+            return workManager.getWorkInfosFlow(workQuery).mapLatest { it.isNotEmpty() }
+        }
```

Comment:
IS this part of the code you modify later on?

If this doesn't change later on, maybe this could have clearer name